components/match2/wikis/mobilelegends/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,651 @@
+	else
+		local suggestedDate = Variables.varDefaultMulti(
+			'matchDate',
+			'Match_date',
+			'tournament_enddate',
+			'tournament_startdate',
+			_EPOCH_TIME
+		)
+		return {
+			date = MatchGroupInput.getInexactDate(suggestedDate),
+			dateexact = false,
+		}
+	end
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'team'))
+	match.type = Logic.emptyOr(match.type, Variables.varDefault('tournament_type'))
+	match.tournament = Logic.emptyOr(match.tournament, Variables.varDefault('tournament_name'))
+	match.tickername = Logic.emptyOr(match.tickername, Variables.varDefault('tournament_tickername'))
+	match.shortname = Logic.emptyOr(match.shortname, Variables.varDefault('tournament_shortname'))
+	match.series = Logic.emptyOr(match.series, Variables.varDefault('tournament_series'))
+	match.icon = Logic.emptyOr(match.icon, Variables.varDefault('tournament_icon'))
+	match.icondark = Logic.emptyOr(match.iconDark, Variables.varDefault("tournament_icondark"))
+	match.liquipediatier = Logic.emptyOr(match.liquipediatier, Variables.varDefault('tournament_liquipediatier'))
+	match.liquipediatiertype = Logic.emptyOr(
+		match.liquipediatiertype,
+		Variables.varDefault('tournament_liquipediatiertype')
+	)
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return match
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = match.stream or {}
+	match.stream = {
+		stream = Logic.emptyOr(match.stream.stream, Variables.varDefault('stream')),
+		twitch = Logic.emptyOr(match.stream.twitch or match.twitch, Variables.varDefault('twitch')),
+		twitch2 = Logic.emptyOr(match.stream.twitch2 or match.twitch2, Variables.varDefault('twitch2')),
+		nimo = Logic.emptyOr(match.stream.nimo or match.nimo, Variables.varDefault('nimo')),
+		trovo = Logic.emptyOr(match.stream.trovo or match.trovo, Variables.varDefault('trovo')),
+		huya = Logic.emptyOr(match.stream.huya or match.huya, Variables.varDefault('huya')),
+		text = Logic.emptyOr(match.stream.text or match.text, Variables.varDefault('text')),
+		separator = Logic.emptyOr(match.stream.separator or match.twitch2, Variables.varDefault('separator')),
+		afreeca = Logic.emptyOr(match.stream.afreeca or match.afreeca, Variables.varDefault('afreeca')),
+		afreecatv = Logic.emptyOr(match.stream.afreecatv or match.afreecatv, Variables.varDefault('afreecatv')),
+		dailymotion = Logic.emptyOr(match.stream.dailymotion or match.dailymotion, Variables.varDefault('dailymotion')),
+		douyu = Logic.emptyOr(match.stream.douyu or match.douyu, Variables.varDefault('douyu')),
+		smashcast = Logic.emptyOr(match.stream.smashcast or match.smashcast, Variables.varDefault('smashcast')),
+		youtube = Logic.emptyOr(match.stream.youtube or match.youtube, Variables.varDefault('youtube')),
+		facebook = Logic.emptyOr(match.stream.facebook or match.facebook, Variables.varDefault('facebook')),
```

Comment:
Feels like this should be moved to a common commons? @hjpalpha thoughts?

Reply:
okay for me to move it into base

Reply:
Was more thinking of an independent module similar to Links. So when a new stream service is added, it needs to be updated in as few places as possible (since not every wiki is using base)

Reply:
I'm agree with this idea too, like a central one in common that list every stream service.

Regarding Trovo/Huya/Text/Separator I have removed those (as Mobilelegends has no use of those, Huya and Trovo probably will be re-add through the central module still, Text and Separator is not)

Reply:
Open an Issue on it and I'll see this conversation as resolved